### PR TITLE
Use local UGW instead of Salus platform

### PR DESCRIPTION
**Pull request by california444**
♻️ Current situation

After Salus has migrated to a new platform and changed the login mechanism, it is currently not possible to use this plugin anymore.

💡 Proposed solution

With this PR the thermostat's data is received via the local UGW instead of requesting data via the Salus cloud platform.

⚙️ Release Notes

The homebridge config for this plugin must be adapted. See README for more information. 

➕ Additional Information

...

Testing

I added some tests with data from my Salus RQ610RF thermostats. I have no idea about other thermostats and if they also work.

### CODE REVIEW

File: src/platformAccessory.ts [hunk 1]
```diff
@@ -39,14 +39,58 @@ export class SalusSQ610Accessory {
       maxValue: 40,
       minStep: 0.5,
     }).onSet(this.onTargetTemperatureSet.bind(this));
+    this.service.getCharacteristic(this.platform.Characteristic.TargetHeatingCoolingState).setProps({
+      validValues: [this.platform.Characteristic.TargetHeatingCoolingState.OFF,
+        this.platform.Characteristic.TargetHeatingCoolingState.HEAT],
```
Comment (bimusiek) [inline src/platformAccessory.ts:44]: Salus thermostats can operate also in "COOLING" mode by shorting the pin in the back or manually setting it up.

**Comment by bimusiek** [comment]
Thanks a lot @california444 This is amazing work, so good to have it working again. 
I have one issue with supporting cooling mode, but this can be fixed in summer as I am not turning on cooling in winter :D 

I will try with my thermostats and if it works, I will merge and release to NPM. Thanks yet again.

**Comment by california444** [comment]
Restricting TargetHeatingCoolingState is maybe not completely thought through from my side ;-)
I am just aware of shortening the pins of KL08RF to change between COOL and HEAT. So my initial thought was that it does not make sense to allow switching to COOL in Home.

For _CurrentHeatingCoolingState_ all modes are supported: OFF/HEAT/COOL.
For _TargetHeatingCoolingState_ we probably need some more testing to figure out what makes sense

**Comment by bimusiek** [comment]
```
For TargetHeatingCoolingState we probably need some more testing to figure out what makes sense
```
As far as I know you can either disable the heating/cooling or enable it. No way to switch between modes as those are not controller by the thermostat. So as long as `CurrentHeatingCoolingState` has support for both heating and cooling, we are good. :)

**Comment by bimusiek** [comment]
Thanks a lot @california444 That's amazing work.

**Pull request merged**